actors/builtin/miner/miner_actor.go

```diff
@@ -397,49 +394,22 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
-		if len(sectorInfos) > 0 {
-			if len(params.Proofs) == 0 {
-				// The miner _was_ supposed to prove something, but didn't.
-				rt.Abortf(exitcode.ErrIllegalArgument, "no proofs submitted in window PoSt for %d sectors", len(sectorInfos))
-			}
-			// Verify the proof.
-			// A failed verification doesn't immediately cause a penalty; the miner can try again.
-			//
-			// This function aborts on failure.
-			verifyWindowedPost(rt, currDeadline.Challenge, sectorInfos, params.Proofs)
+		if len(sectorInfos) == 0 {
+			// Abort verification if all sectors are (now) faults.
+			// It's not rational for a miner to submit a Window PoSt marking *all* non-faulty sectors as skipped,
+			// since that will just cause them to pay a penalty at deadline end that would otherwise be zero
+			// if they had *not* declared them.
+			rt.Abortf(exitcode.ErrIllegalArgument, "cannot prove partitions with no active sectors")
 		}
-
-		// Penalize new skipped faults and retracted recoveries as undeclared faults.
-		// These pay a higher fee than faults declared before the deadline challenge window opened.
-		undeclaredPenaltyPower := postResult.PenaltyPower()
-		undeclaredPenaltyTarget := PledgePenaltyForUndeclaredFault(
-			rewardStats.ThisEpochRewardSmoothed, pwrTotal.QualityAdjPowerSmoothed, undeclaredPenaltyPower.QA)
-		// Subtract the "ongoing" fault fee from the amount charged now, since it will be charged at
-		// the end-of-deadline cron.
-		undeclaredPenaltyTarget = big.Sub(undeclaredPenaltyTarget, PledgePenaltyForDeclaredFault(
-			rewardStats.ThisEpochRewardSmoothed, pwrTotal.QualityAdjPowerSmoothed, undeclaredPenaltyPower.QA,
-		))
-
-		// Penalize recoveries as declared faults (a lower fee than the undeclared, above).
-		// It sounds odd, but because faults are penalized in arrears, at the _end_ of the faulty period, we must
-		// penalize recovered sectors here because they won't be penalized by the end-of-deadline cron for the
-		// immediately-prior faulty period.
-		declaredPenaltyTarget := PledgePenaltyForDeclaredFault(
-			rewardStats.ThisEpochRewardSmoothed, pwrTotal.QualityAdjPowerSmoothed, postResult.RecoveredPower.QA,
-		)
-
-		// Note: We could delay this charge until end of deadline, but that would require more accounting state.
-		totalPenaltyTarget := big.Add(undeclaredPenaltyTarget, declaredPenaltyTarget)
-		err = st.ApplyPenalty(totalPenaltyTarget)
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to apply penalty")
-
-		// Pay penalty
-		vestingPenaltyTotal, balancePenaltyTotal, err := st.RepayPartialDebtInPriorityOrder(store, currEpoch, rt.CurrentBalance())
-		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock penalty for %v", undeclaredPenaltyPower)
-		penaltyTotal = big.Add(vestingPenaltyTotal, balancePenaltyTotal)
-		pledgeDelta = big.Sub(pledgeDelta, vestingPenaltyTotal)
+		if len(params.Proofs) == 0 {
```

Comment:
Instead could we take care of this earlier by changing line 325 to `if len(params.Proofs) != 1`?

Reply:
That should hit this case where the post is actually over active sectors.
And it should not impact correct use because we no longer allow this call to pass when skipping all sectors which seems to be the only valid reason to input 0 proofs.